### ISSUE TITLE
prefer-conditional-expression should also check repeated += and -= assignments

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#3855]
### Bug Report

- __TSLint version__: `5.9.1`
- __TypeScript version__: `2.8.3`
- __Running TSLint via__: gulp-tslint `8.1.3`

#### TypeScript code being linted

```ts
public getSortIcon = (property: string): string => {
    let style = "sortable ";
    if (property === this.sortBy) {
        if (this.sortDirection === "") {
            style += "asc";
        } else {
            style += "desc";
        }
    }
    return style;
}
```

with `tslint.json` configuration:

```json
"prefer-conditional-expression": true
```

#### Actual behavior

No lint violations found

#### Expected behavior
To report a violation for the repeated `+=` assignment within the `if/else` branches and ask to change to code to the following that uses a conditional expression.
```ts
public getSortIcon = (property: string): string => {
    let style = "sortable ";
    if (property === this.sortBy) {
        style += this.sortDirection === "" ? "asc" : "desc";
    }
    return style;
}
```

#### Summary
Basically, it seems that the rule currently only looks for repeated `=` assignments inside the `if/else` branches.  I'd like it to also find repeated `+=` and `-=` assignments and suggest fixes for those as well. This could be an additional rule option to optionally enable, perhaps named `check-increment-decrement`

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
